charts/argo-cd/values.yaml

```diff
@@ -1,184 +1,528 @@
-applicationController:
-  containerPort: 8082
-  servicePort: 8082
-  image:
-    repository: argoproj/argocd
-    tag: v1.2.3
-    pullPolicy: Always
-  volumeMounts: []
-  volumes: []
+## ArgoCD configuration
+## Ref: https://github.com/argoproj/argo-cd
+##
+nameOverride: argocd
 
-server:
-  containerPort: 8080
-  metricsPort: 8083
-  servicePortHttp: 80
-  servicePortHttps: 443
-  serviceAnnotations: {}
+global:
   image:
     repository: argoproj/argocd
     tag: v1.2.3
-    pullPolicy: Always
-  serviceType: ClusterIP
+    imagePullPolicy: IfNotPresent
+
+## Controller
+controller:
+  enabled: true
+  name: application-controller
+
+  image: {}
+  #  repository: argoproj/argocd
+  #  tag: v1.2.1
+  #  imagePullPolicy: IfNotPresent
+
+  ## Argo controller commandline flags
+  args:
+    statusProcessors: "20"
+    operationProcessors: "10"
+
+  ## Argo controller log level
+  logLevel: info
+
+  ## Additional command line arguments to pass to argocd-controller
   extraArgs: []
+
+  ## Annotations to be added to controller pods
+  ##
+  podAnnotations: {}
+
+  ## Labels to be added to controller pods
+  ##
+  podLabels: {}
+
+  ## Configures the controller port
+  containerPort: 8082
+
+  ## Readiness and liveness probes for default backend
+  ## Ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/
+  ##
+  readinessProbe:
+    failureThreshold: 3
+    initialDelaySeconds: 10
+    periodSeconds: 10
+    successThreshold: 1
+    timeoutSeconds: 1
+  livenessProbe:
+    failureThreshold: 3
+    initialDelaySeconds: 10
+    periodSeconds: 10
+    successThreshold: 1
+    timeoutSeconds: 1
+
+  ## Additional volumeMounts to the controller main container.
   volumeMounts: []
-  # - name: ssh-known-hosts
-  #   mountPath: /app/config/ssh
+
+  ## Additional volumes to the controller pod.
   volumes: []
-  # - name: ssh-known-hosts
-  #   configMap:
-  #     name: argocd-ssh-known-hosts-cm
-  annotations: {}
 
-repoServer:
-  containerPort: 8081
-  servicePort: 8081
+  ## Controller service configuration
+  service:
+    annotations: {}
+    labels: {}
+    port: 8082
+
+  ## Node selectors and tolerations for server scheduling to nodes with taints
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+  ##
+  nodeSelector: {}
+  tolerations: {}
+  affinity: {}
+
+  priorityClassName: ""
+
+  resources: {}
+  #  limits:
+  #    cpu: 500m
+  #    memory: 512Mi
+  #  requests:
+  #    cpu: 250m
+  #    memory: 256Mi
+
+  serviceAccount:
+    create: true
+    name: argocd-application-controller
+
+  ## Server metrics controller configuration
+  metrics:
+    enabled: false
+    service:
+      annotations: {}
+      labels: {}
+      servicePort: 8082
+    serviceMonitor:
+      enabled: false
+    #   selector:
+    #     prometheus: kube-prometheus
+    #   namespace: monitoring
+    #   additionalLabels: {}
+
+## Dex
+dex:
+  enabled: true
```

Comment:
This is a really good thing, dex is not critical to argo-cd and many times you don't actually need it. I wonder even if the default should be false, what do you think @alexec ?

Reply:
I'd consider good SSO support one of the defining features of ArgoCD and I'd consider Dex enabled by default a newcomer friendly choice, with the option to move it externally later once your ArgoCD installation needs to scale.